platform/ios/MGLMapboxEvents.m

```diff
@@ -430,9 +435,25 @@ - (void) pushEvent:(NSString *)event withAttributes:(NSDictionary *)attributeDic
     __weak MGLMapboxEvents *weakSelf = self;
 
     dispatch_async(_serialQueue, ^{
+
         MGLMapboxEvents *strongSelf = weakSelf;
+
         if ( ! strongSelf) return;
-        
+
+        if ([MGLEventTypeAppUserTurnstile isEqualToString:event]) {
```

Comment:
I actually like that idea quite a bit.  It further separates the Turnstile tick from the stock Events API conceptually as well as gets rid of the "special cases" which tbh wasn't really sitting well with me.  :+1: